File: ios/Sources/PluginTemplatePlugin/PluginTemplatePlugin.swift
import Foundation
import Capacitor

@objc(PluginTemplatePlugin)
public class PluginTemplatePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "PluginTemplatePlugin"
    public let jsName = "PluginTemplate"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getPluginVersion", returnType: CAPPluginReturnPromise)
    ]

    private let implementation = PluginTemplate()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func getPluginVersion(_ call: CAPPluginCall) {
        call.resolve([
            "version": implementation.getPluginVersion()
        ])
    }
}

File: android/src/main/java/app/capgo/plugintemplate/PluginTemplatePlugin.java
package app.capgo.plugintemplate;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PluginTemplate")
public class PluginTemplatePlugin extends Plugin {

    private final PluginTemplate implementation = new PluginTemplate();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value", "");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void getPluginVersion(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("version", implementation.getPluginVersion());
        call.resolve(ret);
    }
}
